src/btree/bt_sync.c

```diff
@@ -184,11 +237,27 @@ __sync_ref_obsolete_check(WT_SESSION_IMPL *session, WT_REF *ref)
             obsolete = __wt_txn_visible_all(session, newest_stop_txn, newest_stop_durable_ts);
         }
 
-        if (obsolete) {
+        if (obsolete && addr.type == WT_ADDR_LEAF_NO) {
             WT_REF_UNLOCK(ref, WT_REF_DELETED);
             WT_STAT_CONN_DATA_INCR(session, cc_pages_removed);
 
             WT_RET(__wt_page_parent_modify_set(session, ref, true));
+        } else if (obsolete && __sync_should_read_obsolete_page(session, load_throttle)) {
+            /*
+             * This page has overflow items, so we can't just drop it. Instead, read it into memory
+             * and mark it dirty (and ready to evict) so that the next reconciliation will clean it
+             * up. Only take this path for a limited number of pages so as not to churn the cache.
+             */
+            WT_REF_UNLOCK(ref, previous_state);
+            ret = __wt_page_in(session, ref, WT_READ_WONT_NEED | WT_READ_SKIP_DELETED);
+            WT_RET_NOTFOUND_OK(ret);
+            if (ret != WT_NOTFOUND) {
```

Comment:
I'd recommend using `if (ret == 0) {`, I think that's more obvious with respect to the error handling.

Reply:
Fixed in 2fbb617ec.